lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php

```diff
@@ -159,7 +162,10 @@ protected function _getPortableTableColumnDefinition($tableColumn)
                 break;
         }
 
-        if ($this->_platform instanceof MariaDb1027Platform) {
+        if ($type === Types::BOOLEAN) {
+            // As BIT (0b'0' or 0b'1') is used to store boolean values we must cast this default to a php boolean
+            $columnDefault = (bool) bindec(substr($tableColumn['default'], 2, -1));
```

Comment:
A `BIT` column can be nullable. Looks like it's not taken into account.

Reply:
I updated the test to test for `null`. I also removed this part as it's no more relevant.